bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/gmos/GmosScienceAreaGeometry.scala

```diff
@@ -1,86 +1,96 @@
 package edu.gemini.spModel.gemini.gmos
 
+import edu.gemini.shared.util.immutable.ImPolygon
+import edu.gemini.spModel.inst.ScienceAreaGeometry
+import edu.gemini.spModel.gemini.gmos.GmosCommonType._
+import edu.gemini.spModel.gemini.gmos.GmosCommonType.FPUnitMode.{BUILTIN, CUSTOM_MASK}
+import edu.gemini.spModel.obs.context.ObsContext
+
 import java.awt.Shape
-import java.awt.geom.{Rectangle2D, Area}
+import java.awt.geom.{AffineTransform, Rectangle2D, Area}
 
-import edu.gemini.shared.util.immutable.{DefaultImList, ImPolygon}
-import edu.gemini.spModel.inst.ScienceAreaGeometry
-import edu.gemini.spModel.core.Site
-
-import scala.collection.JavaConverters._
-
-class GmosScienceAreaGeometry[I  <: InstGmosCommon[D,F,P,SM],
-                              D  <: Enum[D]  with GmosCommonType.Disperser,
-                              F  <: Enum[F]  with GmosCommonType.Filter,
-                              P  <: Enum[P]  with GmosCommonType.FPUnit,
-                              SM <: Enum[SM] with GmosCommonType.StageMode](inst0: I) extends ScienceAreaGeometry {
-  import GmosScienceAreaGeometry._
-
-  override def geometry: List[Shape] = {
-    Option(inst0).fold(List[Shape]()) { inst =>
-      lazy val width   = inst.getScienceArea()(1)
-      lazy val isSouth = inst.getSite.contains(Site.GS)
-      inst.getFPUnitMode match {
-        case GmosCommonType.FPUnitMode.BUILTIN if inst.isImaging       => List(fov(ImagingFOVSize, ImagingFOVInnerSize))
-        case GmosCommonType.FPUnitMode.BUILTIN if inst.isSpectroscopic => List(longSlitFOV(width))
-        case GmosCommonType.FPUnitMode.BUILTIN if inst.isIFU           => List(ifuFOV(inst.getFPUnit, isSouth))
-        case GmosCommonType.FPUnitMode.BUILTIN if inst.isNS            => List(nsFOV(width))
-        case GmosCommonType.FPUnitMode.CUSTOM_MASK                     => List(fov(MOSFOVSize, MOSFOVInnerSize))
-        case _                                                         => Nil
+import scalaz._
+import Scalaz._
+
+object GmosScienceAreaGeometry extends ScienceAreaGeometry {
+  // Various sizes used in the calculation of the science area in arcsec.
+  val ImagingFovSize          = 330.34
+  val ImagingFovInnerSize     = 131.33 * 2
+
+  val ImagingCenterCcdWidth   = 165.60
+  val ImagingGapWidth         =   3.00
+
+  val MosFovSize              = ImagingFovSize - 2 * 8.05
+  val MOSFovInnerSize         = 139.38 * 2
+
+  val LongSlitFovHeight       = 108.00
+  val LongSlitFovBridgeHeight =   3.20
+
+  override def unadjustedGeometry(ctx: ObsContext): Option[Shape] = {
+    def gmosGeo[P <: FPUnit](mode: FPUnitMode, fpu: P, isSouth: Boolean): Option[Shape] =
```

Comment:
The type parameter isn't doing anything useful here; you can just say `fpu: FPUnit`.